ignite/metrics/precision.py

```diff
@@ -1,60 +1,90 @@
 from __future__ import division
+import warnings
 
 import torch
 
-from ignite.metrics.metric import Metric
+from ignite.metrics.accuracy import _BaseClassification
 from ignite.exceptions import NotComputableError
 from ignite._utils import to_onehot
 
 
-class Precision(Metric):
-    """
-    Calculates precision.
+class _BasePrecisionRecall(_BaseClassification):
 
-    - `update` must receive output of the form `(y_pred, y)`.
-
-    If `average` is True, returns the unweighted average across all classes.
-    Otherwise, returns a tensor with the precision for each class.
-    """
-    def __init__(self, average=False, output_transform=lambda x: x):
-        super(Precision, self).__init__(output_transform)
+    def __init__(self, output_transform=lambda x: x, average=False):
         self._average = average
+        super(_BasePrecisionRecall, self).__init__(output_transform=output_transform)
+
+    def _check_type(self, output):
+        y_pred, y = output
+
+        if y.ndimension() + 1 == y_pred.ndimension():
+            if y_pred.shape[1] == 2:
```

Comment:
@anmolsjoshi could you please recall me why you separate this case and raise a warning ?

Reply:
@vfdev-5 if num_classes=2, it is a binary case that is being fed as a categorical case. I think it'll be helpful to warn the user that only precision for the positive class if calculated in this case.  Because in the case of binary, we shouldn't be average the precision of 0 and 1.

Reply:
But we can also have multiclass 2-classes case which should be computed as N-classes too

Reply:
True, I'll remove the binary_multiclass entirely and treat it as multiclass.

Reply:
I'll take it in new modifications